code/modules/GOAP/cyka_blyat/actions.dm

```diff
@@ -0,0 +1,339 @@
+	if(A.reloads_left)
+		return TRUE
+	else
+		return FALSE
+
+/datum/goap_action/russian/reload/RequiresInRange()
+	return FALSE
+
+/datum/goap_action/russian/reload/Perform(mob/living/simple_animal/hostile/russian/ranged/A)
+	A.say(pick(list("REKLAMA, KRYSHKA MENYA!","Novyy mag, day mne sekundu!","Peregruzochnyy!")))
+	A.ammo_left = A.max_ammo
+	A.reloads_left--
+	action_done = TRUE
+	return TRUE
+
+/datum/goap_action/russian/reload/CheckDone(atom/agent)
+	return action_done
+
+
+/datum/goap_action/russian/medic
+	name = "Heal Allies"
+	cost = 1
+	cooldown = 10
+
+/datum/goap_action/russian/medic/New()
+	..()
+	preconditions = list()
+	preconditions["allyHealed"] = FALSE
+	effects = list()
+	effects["allyHealed"] = TRUE
+
+/datum/goap_action/russian/medic/AdvancedPreconditions(atom/agent, list/worldstate)
+	var/list/viewl = view(10, agent)
+	for(var/mob/living/simple_animal/hostile/russian/ranged/R in viewl)
+		if(!R.stat)
+			if(R.health < (R.maxHealth/2))
+				target = R
+				break
+	return (target != null)
+
+/datum/goap_action/russian/medic/RequiresInRange()
+	return TRUE
+
+/datum/goap_action/russian/medic/Perform(mob/living/simple_animal/hostile/russian/ranged/A)
+	if(target)
+		A.say(pick(list("Isprav'te sebya!","Medik na stsene!","Ostanovite poluchat' vystrel tak mnogo!")))
+		if(prob(1))
+			A.visible_message("[A] hands [target] some vodka! [target] chugs it, feeling rejuvenated!")
+		else
+			A.visible_message(pick(list("[A] applies some libital to [target]!", "[A] bandages some wounds on [target]!", "[A] quickly stitches close a cut on [target]!")))
+		var/mob/living/simple_animal/hostile/russian/ranged/R = target
+		R.health += (R.maxHealth/4)
+		action_done = TRUE
+		return ..()
+	else
+		action_done = TRUE
+		return FALSE
+
+/datum/goap_action/russian/medic/CheckDone(atom/agent)
+	return action_done
+
+/datum/goap_action/russian/resupply
+	name = "Resupply Allies"
+	cost = 1
+	cooldown = 10
+
+/datum/goap_action/russian/resupply/New()
+	..()
+	preconditions = list()
+	preconditions["allyRearmed"] = FALSE
+	effects = list()
+	effects["allyRearmed"] = TRUE
+
+/datum/goap_action/russian/resupply/AdvancedPreconditions(atom/agent, list/worldstate)
+	var/list/viewl = view(10, agent)
+	for(var/mob/living/simple_animal/hostile/russian/ranged/R in viewl)
+		if(!R.stat)
+			if(!R.reloads_left)
+				target = R
+				break
+	return (target != null)
+
+/datum/goap_action/russian/resupply/RequiresInRange()
+	return FALSE
+
+/datum/goap_action/russian/resupply/Perform(mob/living/simple_animal/hostile/russian/ranged/A)
+	var/mob/living/simple_animal/hostile/russian/ranged/RU = target
+	if(target)
+		A.say(pick(list("Voz'mi eto boyepripasy, tovarishch!","Perezagruzite oruzhiye!","Perezagruzite svoy grebanyy idiot!")))
+		A.visible_message("[A] throws a new magazine to [target]!")
+		var/obj/item/russian_reload/RE = new /obj/item/russian_reload(get_turf(AA_MATCH_TARGET_OVERLAYS))
+		RE.throw_at(target, 10, 1)
+		RU.reloads_left = 3
```

Comment:
I feel like this should set their "reloads_left" count to a "max_reloads" value that's defined for each russian mob, for the sake of futureproofing this.

If you know that no future Russian mobs will be intended to never have 3 reloads at a time or will be intended to chew through reloads at a really fast pace for some reason, ignore this comment.

Reply:
Would just be an unused variable right now, and it might stay that way if nobody modifies their max_reloads var later down the line